### DISCUSSION TITLE
How can I learn the Kratos

### DISCUSSION THREAD

**Discussion by suqingji** [KratosMultiphysics/Kratos#11143] [General]
Dear all developers,

I want to learn Kratos for CFD problems. Here are some my quastions.
1. In my expection, I want to know how to add or modify CFD models to simulate some special problem. So where can i find the programming tutorials.
2. If I only want to use C++ to complete the simulate, how do I do that. 
3. Can I create a application used C++ to manipulate all the process by read config file designed by me?

Thanks and best wishes.

**Comment by rubenzorrilla** [comment]
Dear @suqingji,

Let me answer to your queries one by one,

1. Unfortunately, there are no programming tutorials. The most similar thing to a programming tutorial would be the documentation website, which you can find in [here](https://kratosmultiphysics.github.io/Kratos/). In this regard, I note that the documentation website is still a work in progress.
2. You would need to create a new cpp class emulating what is being done by the corresponding analysis stage class (in this case the `fluid_dynamics_analysis.py`) and the corresponding solver. Though it is not the way we would promote, there shouldn't be any technical limitation.
3. Sorry, but I don't fully understand your third question. If you meant to ask about the possibility to create a new custom applications, the answer is yes.
